2018-02-10  Jayant Mehra  <jayant@lnxsrv06.seas.ucla.edu>

        * lib/timespec.h (timespec_cmp):
        Comparing mtimes without subtracting.
        Prevents integer overflow for large numbers.

--- lib/timespec.h      2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h      2018-02-10 08:09:08.868418000 -0800
@@ -45,8 +45,20 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec) {
+    return 1;
+  }
+  else if (a.tv_sec < b.tv_sec) {
+    return -1;
+  }
+  else {
+    if (a.tv_nsec > b.tv_nsec) {
+      return 1;
+    }
+    else if (a.tv_nsec < b.tv_nsec) {
+      return -1;
+    }
+  }
 }

 # if ! HAVE_DECL_NANOSLEEP
